### PR TITLE
Update for final release of macOS 10.14.4 and iOS 12.2

### PR DESCRIPTION
**Pull request by gbrooker**
Update for final release of macOS 10.14.4 and iOS 12.2, which include support for the Television accessory.

### CODE REVIEW

**Comment by gbrooker** [comment]
This is not compiling because I included a minor change in `hap-server` which is required by swift 4.2. Is it now possible to change the build to swift 4.2 or are there still issues on linux meaning you need to stick to swift 4.1 ?

**Comment by Bouke** [comment]
I'm using Swift 4.2 now on my Raspberry, so I'm fine with dropping 4.1. 👍 It would be great if we could explicitly target 4.2 and 5.0 from now on.

File: Sources/hap-server/main.swift [hunk 1]
```diff
@@ -115,7 +115,7 @@ withExtendedLifetime([delegate]) {
         RunLoop.main.run(until: Date(timeIntervalSinceNow: 10))
     } else {
         while keepRunning {
-            RunLoop.current.run(mode: .default, before: Date.distantFuture)
+            RunLoop.current.run()
```
Comment (Bouke) [inline Sources/hap-server/main.swift]: This is not equivalent [see: run() RunLoop](https://developer.apple.com/documentation/foundation/runloop/1412430-run):

> If you want the run loop to terminate, you shouldn't use this method. Instead, use one of the other run methods and also check other arbitrary conditions of your own, in a loop. A simple example would be: (...)

Comment (gbrooker) [reply Sources/hap-server/main.swift]: You are right, I've just reverted this change

**Comment by Bouke** [comment]
FYI: I've started a branch to see what's needed to support Swift 5. See also #84.

**Comment by gbrooker** [comment]
So strange, it seems osx and linux swift 4.2 do not have the same definition of RunLoop

**Comment by Bouke** [comment]
Yeah that’s the big downside to Swift Foundation, it’s functionality wise not at par with the OSX counterpart and API wise also lagging behind on changes in the OSX counterpart.

**Comment by Bouke** [comment]
I've merged Swift 5 support, including a fix for the Foundation naming conflict.

**Comment by gbrooker** [comment]
This should now be ready to be merged

**Pull request merged**